File: SQLQuery.php
<?php

/**
 * This class lets you define your SQL queries in separate logic files 
 * and then comfortably import them in right place in your code. 
 * 
 * Keeps code clean and maintainable.
 *
 * @author jozef_cipa
 *
 */
class SQLQuery{

    /**
     * @var array Array of files in directory
     */
    private $sqlFiles;

    /**
     * @var string Directory path
     */
    private static $dir = './sql';

	/**
	 * @var string Name of part from sql file
	 */
    private $sqlPart;

    private function __construct($sqlPart)
    {
        $this->sqlFiles = glob(self::$dir  . "/*.sql");
        $this->sqlPart = $sqlPart;
    }

    /**
     * Change default sql directory path
     * @param string $dir New path
     */
    public static function setSqlDir($dir){
        self::$dir = $dir;
    }

    /**
     * Return new object and set sql part
     * @param  string $sqlPart Name of part from sql file
     * @return SQLQuery Return new object
     */
    public static function import($sqlPart = '*'){
        
        return new self($sqlPart);
    }

    /**
     * Return whole sql string or its part from given filename
     * @param  string $filename Name of sql file
     * @return Return query string
     */
    public function from($filename)
    {

        $filename .= '.sql';
        
        if (!$this->fileExists($filename))
            throw new Exception("File $filename doesn't exist in directory " . self::$dir);
        
        if ($this->sqlPart == '*')
            return $this->getSQL($filename);
        else
            return $this->getSQLPart($this->sqlPart, $filename);

    }

    /**
     * Check if given filename exists in sql directory
     * @param string $filename Name of sql file
     * @return boolean 
     */
    private function fileExists($filename){
        
        $files = array_map(function($filename){
            return basename($filename);
        }, $this->sqlFiles);
        
        return in_array($filename, $files);
    }

    /**
     * Return content of file
     * @param  string $filename File name 
     * @return string Return content of file
     */
    private function readFile($filename){
        return file_get_contents(self::$dir . '/' . $filename);
    }


    /**
     * Return whole sql string 
     *
     * @param $filename string File name
     * @return string Return whole sql string
     */
    private function getSQL($filename){
        return $this->readFile($filename);
    }

    /**
     * Return sql part from file by $sqlPartName
     * @param  string $sqlPartName Name of sql part
     * @param  string $filename File name
     * @return string Return sql string
     */
    private function getSQLPart($sqlPartName, $filename){

        $sqlFile = $this->getSQL($filename);

        $parts = preg_split("/--@SQLName(.*)\\n/", $sqlFile);

        //remove first empty string
        array_shift($parts);

        //remove spaces from start and end 
        $parts = array_map(function($item){
            return trim($item);
        }, $parts);

        preg_match_all("/--@SQLName (.*)\\n/", $sqlFile, $partsNames);

        $partIndexByName = array_search($sqlPartName, $partsNames[1]);

        $sql = $parts[$partIndexByName];
        
        if(!$sql)
            throw new Exception("SQL part with given name: $sqlPartName doesn't exist in $filename.");
        
        return $sql;
    }
}

File: SQLQuery.py
# -*- coding: utf-8 -*-

import glob
from os.path import basename
import re
__author__ = "jozef_cipa"


class SQLQuery:
    """
        This class lets you define your SQL queries in separate logic files
        and then comfortably import them in right place in your code.
        Keeps code clean and maintainable.
    """

    _dir = "./sql"

    def __init__(self, sql_part):
        # List of files in directory
        self.sql_files = glob.glob(SQLQuery._dir + "/*.sql")

        # Name of part from sql file
        self.sql_part = sql_part

    @classmethod
    def set_sql_dir(cls, dir_path):
        """ Change default sql directory path """

        cls._dir = dir_path

    @classmethod
    def import_sql(cls, sql_part='*'):
        """ Return new object and set sql part """

        return SQLQuery(sql_part)

    def from_file(self, filename):
        """ Return whole sql string or its part from given filename """

        filename += '.sql'

        if not self._file_exists(filename):
            raise Exception("File " + filename + " doesn't exists in directory" + SQLQuery._dir)

        if self.sql_part == '*':
            return self.get_sql(filename)
        else:
            return self.get_sql_part(self.sql_part, filename)

    def _file_exists(self, filename):
        """ Check if given filename exists in sql directory """

        files = map(lambda file: basename(file), self.sql_files)

        return filename in files

    def _read_file(self, filename):
        """ Return content of file """

        with open(SQLQuery._dir + '/' + filename) as f:
            return f.read()

    def get_sql(self, filename):
        """ Return whole sql string """

        return self._read_file(filename)

    def get_sql_part(self, sql_part_name, filename):
        """ Return sql part from file by sql_part_name """

        sql_file = self.get_sql(filename)

        parts = re.split('--@SQLName(.*)\\n', sql_file)  # return list ['name', 'statement', 'name2', 'statement2']
        parts.pop(0)  # remove first empty string

        parts = map(lambda item: item.strip(), parts)  # remove spaces from start and end

        sql_parts = []

        sql_names = parts[0::2]
        sql_statements = parts[1::2]

        if len(sql_names) is not len(sql_statements):
            raise Exception("Wrongs sql file format")

        for i in range(len(sql_names)):
            sql_parts.append({sql_names[i]: sql_statements[i]})

        for sql_part in sql_parts:
            if sql_part_name in sql_part:
                return sql_part[sql_part_name]
        else:
            raise Exception("SQL part with given name: " + sql_part_name + " doesn't exist in " + filename)
